313 design a system for guidance users to be able to edit data like courses or survey

RichardM1337 opened the pull request:
- courseList 
- courseTable based on studentTable (no flags)
- surveyList 
- editCourse
- editSurvey
- made sort button dynamic that takes in array
- added everything to navbar

Samuel-Kipnis committed:
create CourseList. create dynamic course form opened on clicking on a class

Samuel-Kipnis committed:
r

RichardM1337 committed:
Update duplicates

might have removed the wrong functions

RichardM1337 committed:
fixed code

Samuel-Kipnis committed:
default value, submit button

Samuel-Kipnis committed:
e

Samuel-Kipnis committed:
e

RichardM1337 committed:
fixed student table, added navbar courselist

RichardM1337 committed:
Merge branch '313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey' of https://github.com/sitechtimes/course-selection-frontend-v2 into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

RichardM1337 committed:
many changes

introduced a CourseTable that emulates studentTable wtihout flags, for use wtih courses. CourseList uses this table

changed sortButton to be dynamic by allowing the array doing the sorting to be a prop (which is why student list is also changed). SortArray interface currently unused but maybe enforced later

introduced a getCourses function in the stores with a currently unused courseList

what needs to be done:
more sorting
possibly remove flags from courses, or reuse them for a separate purpose..? this requires making ChangeFlag dynamic

RichardM1337 committed:
hotfixe to coursetable

removed flags
fixed ap and honors

RichardM1337 committed:
added grades, fixed sorts

Samuel-Kipnis committed:
w

Samuel-Kipnis committed:
Merge branch '313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey' of https://github.com/sitechtimes/course-selection-frontend-v2 into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

Samuel-Kipnis committed:
e

RichardM1337 committed:
survey list done, progress made on editing

logic may have to be chacnged, each question has an id so I can iterate throguh survey[route.params.grade]

RichardM1337 committed:
fixing frontend editsurvey

temporarily turned off sruvey submission navbar

RichardM1337 committed:
hotfixes ,type changes. unused imports GONE!!!

RichardM1337 committed:
removed excess

RichardM1337 committed:
typescript issues

EditCourse string declaration may not be a good idea

kennething commented on src/components/Guidance/CourseTable.vue:
```diff
@@ -0,0 +1,60 @@
+
+const props = defineProps<{ courses: Course[] }>();
+function displayGrades(course: Course) {
+ const grades: string[] = [];
```
what is this 1 spaced ident??

kennething commented on src/components/Guidance/CourseTable.vue:
```diff
@@ -0,0 +1,60 @@
+function displayGrades(course: Course) {
+ const grades: string[] = [];
+ if (course.freshman) grades.push("Freshman");
+  if (course.sophomore) grades.push("Sophomore");
```
why is there an indent here??

kennething commented on src/components/Guidance/CourseTable.vue:
```diff
@@ -0,0 +1,60 @@
+</script>
+
+<style scoped>
+.parent:hover .child {
```
unused chatgpt css??

RichardM1337 replied:
code from studenttable.vue, most likely to handle :hover. i can change it tho

kennething replied:
well neither this nor studentTable have anything with a .parent or .child class so definitely get rid of it in both

kennething commented on src/components/Guidance/SortButton.vue:
```diff
@@ -29,7 +29,8 @@
 <script setup lang="ts">
 import { ref } from "vue";
-
+import {SortArray} from '../../types/interface';
```
prettier doesnt format imports like this?? its supposed to be `{ SortArray }`??

kennething commented on src/components/Navbar/NavBar.vue:
```diff
@@ -147,7 +153,8 @@ const surveyStore = useSurveyStore();
 const save = ref("Save");
 
-const viewingSurvey = () => router.currentRoute.value.path.includes("survey");
+/* const viewingSurvey = () => router.currentRoute.value.path.includes("survey"); */
```
forgot to uncomment this?

kennething commented on src/router/index.ts:
```diff
@@ -1,133 +1,148 @@
-      ],
-    },
-  ],
+	history: createWebHistory(import.meta.env.BASE_URL),
```
8 spaced indents now????

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+            <input
+              v-model="input"
+              placeholder="Search by name or email"
+              @input="
```
ur using the outdated method, use the new one https://vuejs.org/api/sfc-script-setup.html#definemodel

kennething commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,122 @@
+  return grades.join(", ");
+}
+
+async function fetchData(url: string, method?: string, body?: any) {
```
why are you copy pasting the functions already in the stores ??

kennething commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,122 @@
+  if (method) {
+    options["method"] = method;
+    options["headers"] = { "Content-Type": "application/json" };
+    options["body"] = body;
```
you didnt even copy it right?? why isn't this json.stringify(body) ??

kennething commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,122 @@
+import { useRoute, useRouter } from "vue-router";
+import {Course} from '../types/interface';
+
+const alteredCourse = ref<Course>({} as Course);
```
why not just ref<Course>() and handle the undefined? clearly it starts off as an empty object anyways

kennething commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,112 @@
+        "
+      >
+      <div class="flex flex-col p-3">
+  <label class="text-2xl pb-1 font-bold" for="dueDate">Due Date</label>
```
formatting was invented in 1450
people in 1449:

RichardM1337 committed:
kenneth fixes

formatting
using userStore where necessary
replaced viewingSurvey to only have "save" and "exit" on nav bar when user is student
added getCourses, getSurveys, and fetchData to user.ts exports
replaced alteredSurvey and alteredCourse with computed values to handle undefined (thx Kenneth)

whydoesthecodenotwork commented on src/components/Guidance/CourseTable.vue:
```diff
@@ -0,0 +1,60 @@
+        <tr class="hover:bg-other-g transition-all">
+          <td
+            class="p-4 hover:cursor-pointer"
+            @click="router.push(`/guidance/editcourse/${course.id}`)"
```
"we have routerlink at home"

RichardM1337 replied:
hey, i'm having trouble implementing this.
wrapping the button in a routerlink prevents @submit.prevent from being activated.
should i instead add the put request in the button's @submit?

whydoesthecodenotwork replied:
i'm also confused
there's no button or @submit.prevent or put request in this file
what file are these in?

whydoesthecodenotwork commented on src/components/Navbar/NavBar.vue:
```diff
@@ -47,6 +47,12 @@
+        <p class="text-base hover:text-gray-500">Surveys</p>
+      </RouterLink>
+      <RouterLink id="link" to="/guidance/courselist">
+        <p class="text-base hover:text-gray-5000">Courses</p>
```
that's a lot of text-gray

whydoesthecodenotwork commented on src/components/Navbar/NavBar.vue:
```diff
@@ -47,6 +47,12 @@
+      <RouterLink to="/guidance/surveylist">
+        <p class="text-base hover:text-gray-500">Surveys</p>
+      </RouterLink>
+      <RouterLink id="link" to="/guidance/courselist">
```
there are duplicate ids in this file
(why are there even ids on these routerlinks??)

whydoesthecodenotwork commented on src/stores/user.ts:
```diff
@@ -145,6 +151,18 @@ export const useUserStore = defineStore("user", () => {
+    const res = await fetchData("guidance/survey/");
+    if (!res.ok) return await res.json();
+    const data = await res.json();
+    return data;
```
? what does checking for `(!res.ok)` do
it looks like you're returning the same thing either way..

whydoesthecodenotwork commented on src/types/interface.ts:
why are you using tabs here but 2 spaces elsewhere

whydoesthecodenotwork replied:
and `"` and `'`

whydoesthecodenotwork commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+        ❯❯
+      </button>
+    </div>
+    <h5 class="mt-4">
```
is there an h1 or h2 or h3 or h4 on this page (i am too lazy to clone and check the actual page, but uhhhh try not to skip heading levels)

whydoesthecodenotwork commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+  if (sortBy === "freshman" || sortBy === "sophomore" || sortBy === "junior" || sortBy === "senior")
+    return courses.filter((course) => course[sortBy] == true);
+  if (sortBy === "honors" || sortBy === "ap") {
+    return courses.filter((course) => course[sortBy] == true);
```
could `===` be used instead of `==` here

whydoesthecodenotwork commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+  else if (increment < 0) startIndex.value -= pageCapacity;
+  else return;
+  updatePagination(currentPage.value);
+}
```
increment is only ever 1 or -1 here, so. you can just multiply increment and page capacity
also not sure when increment would ever be 0?
```ts
function changePage(increment: number) {
  currentPage.value += increment;
  startIndex.value += increment * pageCapacity;
  updatePagination(currentPage.value);
}
```

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+
+const viewAll = ref(false);
+const input = ref("");
+const sortBy = ref("az");
```
make this `typeof menuArray[number]["sortBy"]` to ensure its only a `sortBy` value from `menuArray`
also put `as const` after the `]` but before the `;` for `menuArray`

whydoesthecodenotwork commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+          />
+          <select
+            name=""
+            id=""
```
should the key be in here

whydoesthecodenotwork commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+            v-else-if="key == 'description'"
+          />
+          <select
+            name=""
```
do you need to specify an empty name attribute

whydoesthecodenotwork commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+          </select>
+        </div>
+        <div>
+          <button class="p-5 border-2 border-black bg-white hover:bg-other-g">
```
does this need `type="submit"`

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+
+const sortedAndFiltered = computed(() => {
+  try {
+    return applyFilters(sortBy.value, input.value);
```
logically this should never throw an error ??
are you space-proofing this code from cosmic bit flips ??

whydoesthecodenotwork commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+
+async function findCourse() {
+  let courses = await userStore.getCourses();
+  return courses.find((course: Course) => course.id == Number(courseId));
```
consider strict equality `===`

whydoesthecodenotwork commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+}
+
+async function findCourse() {
+  let courses = await userStore.getCourses();
```
can this be const

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,197 @@
+});
+
+function filterByCategory(courses: Course[], sortBy: string) {
+  if (!Array.isArray(courses)) return [];
```
when is courses not an array ??

RichardM1337 committed:
kenneth!

routerlink

used definemodel

kennething commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+  return courses.find((course: Course) => course.id == Number(courseId));
+}
+
+const course = ref<Course>({} as Course);
```
1. the `<Course>` is unnecessary because you're using `as Course` in the initial declaration already
2. don't use `as Course` on an empty object when it is clearly not a full `Course` object ?? just do `ref<Course>()` handle the undefined case

whydoesthecodenotwork commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,98 @@
+            id="dueDate"
+            class="border-2 border-black rounded-lg p-2 mb-4"
+          />
+          <input
```
this input has no associated label
if you don't want a visible label, consider using a visually hidden label
`<label for="dueTime" class="sr-only">Due Time</label>`
(sr-only = screen reader only. think of the blind people!)

kennething commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,98 @@
+        @submit.prevent="
+          () => {
+            const isoString = new Date(
+              `${dueDateDate}T${dueDateTime}`
```
what you did:
1. manually create an ISO string
2. create a date object from the manually created ISO string
3. turn the date object (created from a manually created ISO string) into an ISO string

whydoesthecodenotwork commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,98 @@
+          />
+        </div>
+        <div>
+          <button class="p-5 border-2 border-black bg-white hover:bg-other-g">
```
does this need `type="submit"`

kennething commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,98 @@
+const userStore = useUserStore();
+
+async function findSurvey() {
+  let surveys = await userStore.getSurveys();
```
unnecessary let

whydoesthecodenotwork commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,98 @@
+}
+
+const survey = ref<Survey>({} as Survey);
+let surveyKeys = [] as (keyof Survey)[];
```
i have no idea if this actually works but it feels cursed
you're using v-for on a normal non-reactive array
(same with courseKeys earlier)

if it works it works i guess??

kennething commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,97 @@
+
+async function findSurvey() {
+  const surveys = await userStore.getSurveys();
+  return surveys.find((survey: Survey) => survey.grade == Number(surveyGrade));
```
===

kennething commented on src/views/EditSurvey.vue:
```diff
@@ -0,0 +1,97 @@
+}
+
+const survey = ref<Survey>({} as Survey);
+let surveyKeys = [] as (keyof Survey)[];
```
what is this used for ??

whydoesthecodenotwork replied:
oh yeah courseKeys gets v-ford but surveyKeys is just never used

RichardM1337 replied:
sorry guys. courseKeys is iterating through course data and courseKeys is used to ignore certain ones. surveyKeys was unnecessary because im iterating thru question :(

whydoesthecodenotwork commented on src/views/SurveyList.vue:
```diff
@@ -0,0 +1,60 @@
+  </Suspense>
+</template>
+  
+  <script setup lang="ts">
```
yeah ouch!

whydoesthecodenotwork replied:
do you have prettier setup and set to format on save

kennething commented on src/views/SurveyList.vue:
```diff
@@ -0,0 +1,60 @@
+  <Suspense>
+    <template #fallback>
+      <div
+        class="fixed top-[50%] left-[50%] translate-x-[-50%] translate-y-[-50%] text-2xl font-bold"
```
top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2

BSelyomin replied:
That's fucking wild

whydoesthecodenotwork replied:
this hexes me.

whydoesthecodenotwork commented on src/views/SurveyList.vue:
```diff
@@ -0,0 +1,60 @@
+        :to="'/guidance/editsurvey/' + survey.grade"
+        class="bg-primary-g border-black border-2 drop-shadow-lg rounded-lg p-6 hover:drop-shadow-xl hover:bg-other-g transition-all duration-300"
+      >
+        <h2 class="text-2xl font-bold">{{ survey.grade }}th Grade</h2>
```
i don't know if this skips a heading level but if it does then yeah ouch

kennething commented on src/views/SurveyList.vue:
```diff
@@ -0,0 +1,60 @@
+}
+
+async function getSurveys() {
+  const response: Response = await fetchData("guidance/survey", "GET");
```
fetch (and fetchData by association) already default to GET

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,185 @@
+const pageCapacity = 10;
+const currentChunk = ref(1);
+const pagesPerChunk = 10;
+const menuArray = [
```
needs to be `as const`

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,185 @@
+import { Course } from "../types/interface";
+import { useUserStore } from "../stores/user";
+import { ref, computed, watch, onMounted } from "vue";
+document.title = "Course List | SITHS Course Selection";
```
why do you put document.title in a different place in every file

RichardM1337 replied:
Where should I put them

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,185 @@
+  </div>
+</template>
+
+<script setup lang="ts">
```
add some line breaks between unrelated declarations geez

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,185 @@
+const userStore = useUserStore();
+const courses = ref<Course[]>([]);
+const viewAll = ref(false);
+const input = defineModel({ type: String });
```
do `defineModel<string>()` instead https://vuejs.org/api/sfc-script-setup.html#usage-with-typescript

kennething commented on src/views/CourseList.vue:
```diff
@@ -0,0 +1,185 @@
+        <Sort
+          class="mr-0"
+          :menu-array="menuArray"
+          @filter-selected="(filter:string) => (sortBy = filter)"
```
define the emit type in `Sort` so you don't need to do `filter: string`

kennething commented on src/views/EditCourse.vue:
```diff
@@ -0,0 +1,106 @@
+            :placeholder="course[key]"
+            :id="key"
+            class="border-2 border-black rounded-lg p-2 mb-4 form-textarea textarea-xl"
+            v-else-if="key == 'description'"
```
===

kennething commented on src/views/SurveyList.vue:
```diff
@@ -0,0 +1,60 @@
+  </Suspense>
+</template>
+  
+  <script setup lang="ts">
```
evil indent

RichardM1337 committed:
pr changes + hours added

routerlink button pronblem needs work

RichardM1337 committed:
have mercy

RichardM1337 committed:
removed mention of Date object to maintain timzeone

RichardM1337 committed:
hotfix

RichardM1337 committed:
routerpuush

Shatterdest commented:
ochen typescript errors

RichardM1337 commented:
![image](https://github.com/user-attachments/assets/99219f65-1ec1-4a72-8c11-4f2692bb21b7)

Shatterdest committed:
f

Shatterdest committed:
interesting

Shatterdest committed:
Update EditCourse.vue

RichardM1337 committed:
z

- make checkbox darker
- added label for

RichardM1337 committed:
css

where is my date popup

RichardM1337 committed:
css

can sort key array by bool then string to  be able to create a more specific area for checkboxes

RichardM1337 committed:
css

todo:
datepopup
maybe sort keys in editcourse by data type to put checkboxes all at da end with one big chart -- may require awaiting or sumn cuz right now they are in order

also does rows even work mane

RichardM1337 committed:
css

todo:
remove excess
conform text color to waves
fix placeholder
aria labels

RichardM1337 committed:
css

editCourse -- placeholder hotfix by defining altered Course separately
survey -css

may have to include flex-row in fieldset to fill empty checkbox space

RichardM1337 committed:
css fixes

RichardM1337 committed:
questionstable being worked on

RichardM1337 committed:
question editing add

question table works, added question editing and some sorting

RichardM1337 committed:
lots of quesiton editing

this code is very ugly. feel free to use chat gpt to tell me what is wrong with m code. chat gpt told me to use pick. DOnt blame me. I am a great coder. I getting a job at faang. Trust me.

RichardM1337 committed:
added change.prevent.

baklcend still needs the put request fixed howeevr

RichardM1337 committed:
changes

status now works, quesiton type doesnt. Also foreignKey in django implies the forntend only needs to submit an id and the backend should serialize it :) mus change

RichardM1337 committed:
Update EditQuestion.vue

RichardM1337 committed:
Update EditQuestion.vue

quick bugfix that allows referenced classes to be properly grabbed in the backend

RichardM1337 committed:
added functionality for placeholder

rpetty uggly but it seems difficult to make a placeholder for vue select. Could use v-model but no optional chaining and doesnt seem to work reliably

RichardM1337 committed:
css and random fixes

todo list
- surveys by grade question adder
- css
- potentially adding changes to options
- media quering this huge table

RichardM1337 committed:
added proper functionality for choosing survey questions

todo:
- make it some that available questions do not include survey questions
- actual submitting functionality (which will have lots of checkers and such -- also gotta see if questions being passed to backend to be added to survey need to be serialized)

RichardM1337 committed:
fixed filtering -- adding survey quetions works

the form automatically interprets the surveyQuetstions of models of the alteredQuestion I think, but I still have to wrok on removing questions

RichardM1337 committed:
media queriing

RichardM1337 committed:
css fixes and bugfix

RichardM1337 committed:
ts hotfix

RichardM1337 commented:
courseList
courseTable based on studentTable (no flags)
editCourse
surveys & questionList
questionTable based on studentTable (no flags)
editQuestion
editSurvey
made sort button dynamic that takes in array
added everything to navbar
appropriate userstores to grab surveys, quesestions
classreferenced type based on Pick

RichardM1337 committed:
Merge branch 'dev' into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

RichardM1337 committed:
Merge branch 'dev' into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

RichardM1337 committed:
Update package-lock.json

whydoesthecodenotwork commented:
I have been summoned

kennething commented:
oh i am 6 months late to this review

RichardM1337 commented:
hooold on twin atp it's legacy code, i think i have to reuse it in another branch

RichardM1337 commented:
spare me kenneth

RichardM1337 committed:
Functional website, but ugly

tinazhen08 committed:
Update package-lock.json

Shatterdest committed:
Update package-lock.json

Shatterdest committed:
Merge branch '313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey' of https://github.com/sitechtimes/course-selection-frontend-v2 into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

tinazhen08 committed:
Merge branch 'dev' into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

tinazhen08 committed:
Merge branch 'dev' into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

tinazhen08 committed:
fix view survey button

RichardM1337 committed:
Frontend implementation of making new Survey questions

This has no CSS

RichardM1337 committed:
Update SurveyList.vue

emergency hotfix to make the button visible

tinazhen08 committed:
fix filter to sort by last name

RichardM1337 committed:
DeleteQuestion

tinazhen08 committed:
change drag and drop for surveys

fix css tmr

tinazhen08 committed:
Merge branch '313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey' of https://github.com/sitechtimes/course-selection-frontend-v2 into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

tinazhen08 committed:
change student table css

RichardM1337 committed:
hotfix to how courses are type declared

let me know if this causes any issues

RichardM1337 committed:
Update CourseList.vue

RichardM1337 committed:
Update user.ts

please let me know if this causes any issues, im trying to find errors with netlify building

RichardM1337 committed:
Update user.ts

tinazhen08 committed:
finished css

just need to fix flags on student list

tinazhen08 committed:
flag buttons and new question card

Shatterdest committed:
f

Shatterdest committed:
Merge branch '313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey' of https://github.com/sitechtimes/course-selection-frontend-v2 into 313-design-a-system-for-guidance-users-to-be-able-to-edit-data-like-courses-or-survey

Shatterdest committed:
Update user.ts

tinazhen08 committed:
Update ChangeFlag.vue

RichardM1337 committed:
Update package-lock.json

high severity vulnerability

RichardM1337 committed:
Vue and lint changes

RichardM1337 committed:
converted course.subjects into a dropdown

RichardM1337 committed:
Update interface.ts

RichardM1337 committed:
Temporary CSS change to reviewed surveys

the radios and checkboxes were barely visible. reverse if needed

RichardM1337 committed:
Update SortButton.vue

text width hotfix

RichardM1337 committed:
Update EditSurvey.vue

added onclick to edit survey

RichardM1337 committed:
creation questions now works with dorpdown

RichardM1337 committed:
Update EditQuestion.vue

now each element can be removed as seen fit and everything is changed as a temporaryOptions sot hat refreshes o anyhing dont just randomly delete data. COol!

RichardM1337 commented:
Happy 1 year anniversary of this repo
